### ISSUE TITLE
CAMPD 508 remediation Query Data

### ISSUE THREAD

**Issue by alangmaid** [US-EPA-CAMD/PI-1#6361]
1) On page Data > Query Data > Filters:
Vintage Year (Under Allowance, Transactions) has no accessible name.  Returns error "Improper use of [aria-labelledby]: Referenced id not found.

2) On page Data > Query Data > Preview Data: 
The tagging on the sort buttons on the data table are reversed from their function.  Rather than reading what the button will do, it reads the current sort state.  When a column is already in ascending order, the button reads "sort by ascending." Clicking the button changes the sort to descending order.

![Image](https://github.com/user-attachments/assets/a21e62f2-6d88-4dc4-9335-809f81d2dfd1)

[CAMPD.Report.docx](https://github.com/user-attachments/files/16617480/CAMPD.Report.docx)

**Comment by yonatan-dp** [comment]
@alangmaid I am not able to reproduce defect #1 above. I am able to see the aria-label. See screenshot below.

![image](https://github.com/user-attachments/assets/cf64e198-5c47-400f-a44f-ce6f3fd333c6)

**Comment by alangmaid** [comment]
@szintgraff please see @yonatan-dp comment above. As reported by @yonatan-dp item1 appears to be a non-issue.

**Comment by szintgraff** [comment]
@alangmaid and @yonatan-dp For Item 1 above: Can you ensure that both locations that say vintage year have the correct functionality? 
![image](https://github.com/user-attachments/assets/335816f4-16e8-4f4b-8e8a-2eb8090795e2)

**Comment by alangmaid** [comment]
@szintgraff We'll follow-up during DSU tomorrow.

**Comment by yonatan-dp** [comment]
Updated Comment: Please see note added on August 24 for the latest on this. After the original comment below was added, I was able to reproduce and fix this issue.

Original Comment:
I used both the ANDI tool and the WAVE tool to replicate the reported issue (item #1 indicated in the description) but I was not able to reproduce the issue.  Here is a screenshot of the WAVE tool

![image](https://github.com/user-attachments/assets/967fd387-43f5-4db1-82b9-876421d84964)

**Comment by szintgraff** [comment]
@yonatan-dp Just to confirm you used the WAVE and ANDI tools to check the accessibility functionality in both locations that say "Vintage Year"  (I highlighted the locations above) and saw they are functioning correctly?

**Comment by yonatan-dp** [comment]
@szintgraff, @mxtomoto1 I did another round of testing and I am now able to reproduce the problem you indicated both in Firefox and Chrome.  The ANDI tool required refreshing once on the page is completely loaded in order to show the error. Here is a screenshot

![image](https://github.com/user-attachments/assets/141c40c5-7729-4525-803f-76448fa746cf)

I have fixed the issue.

**Comment by mxtomoto1** [comment]
Tested on the UI for CAMPD-stg. User is able to filter by vintage year, but the arrow filter button for ASC / DESC indicates the current sort state.

@szintgraff @yonatan-dp are these issues meant to be tested on the WAVE / ANDI tools? If so, how can I get access to them?

Edit: I am having trouble accessing ANDI on the page for data download and may need some assistance.

**Comment by szintgraff** [comment]
Tested the vintage year sorting and it looks good. I can see the code where the ARIA text was changed to announce the new sorting direction (item 2), but I can't find where the change happened for item 2? @alangmaid @mark-hayward-erg 

@mxtomoto1 Your last comment says you are having trouble accessing ANDI for testing. Can you update what was decided and why this is moved to UAT if you were unable to test? I also don't have those tools so I am just confirming that item 1 was updated.

**Comment by alangmaid** [comment]
@szintgraff we can confirm that item 2 has been addressed without using ANDI. @mxtomoto1 Please upload a screen capture as evidence that item 2 has been addressed.

**Comment by szintgraff** [comment]
@alangmaid Apologies I meant item 1. I can't find where the code was changed for item 1.

**Comment by yonatan-dp** [comment]
@szintgraff, Please look at the following two PRs for the changes.  

- https://github.com/US-EPA-CAMD/easey-campd-ui/pull/650/files
- https://github.com/US-EPA-CAMD/easey-campd-ui/pull/649/files

**Comment by mxtomoto1** [comment]
Screenshot showing page's behavior for item 2 (see Transaction ID column).

<img width="969" alt="asc" src="https://github.com/user-attachments/assets/d3fdbfd7-90e2-449a-abf0-6276024e6e59">
